src/session/session.c

```diff
@@ -1043,6 +1054,13 @@ main (int argc,
       status = spawn_and_wait (bridge_argv, env, NULL, -1, pwd->pw_uid, pwd->pw_gid);
     }
 
+  if (WIFEXITED (status) && WEXITSTATUS (status) == EXIT_NOT_FOUND)
+    {
+      debug ("bridge exited with status %d, no-cockpit", status);
+      exit_init_problem (EXIT_NOT_FOUND);
```

Comment:
This will exit immediately without doing the finalization actions below, such as calling `pam_end` and `gss_release_cread`. Is that OK?

Right now I am thinking that we should not call `exit_init_problem` here (and should not change `exit_init_problem` to handle `EXIT_NOT_FOUND`).  Instead, what about making a new function to send the right `init` message for the "not found" condition, call it here, and then continue with the rest of this function?

Reply:
Right, good idea -- I reworked this accordingly. Seems to work with the tests in this PR, I also re-test a rebased #16808.